src/include/graph/graph.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "common/types/types.h"
+
+namespace kuzu {
+
+namespace graph {
+
+/*
+ * Graph is implemented with the following assumptions
+ * - single node and edge table.
+ * - topology only, i.e. does not track property.
+ * */
+class Graph {
+public:
+    Graph() = default;
+    virtual ~Graph() = default;
+
+    // TODO: we may not need the following interface.
+    virtual common::table_id_t getNodeTableID() = 0;
+
+    // Get total number of nodes in the graph.
+    virtual common::offset_t getNumNodes() = 0;
+    // Get total number of edges in the graph.
+    virtual common::offset_t getNumEdges() = 0;
+
+    // Scan neighbors using forward index.
+    // Return total number of neighbors.
+    virtual uint64_t scanNbrFwd(common::offset_t offset) = 0;
```

Comment:
You are using scanNbrFwd and getNbr() as a very hacky way to implement an iterator over neighbors of a node. I would remove them and force the algorithms to use getNbrs(), which serves the same purpose. We don't care about efficiency now anyways. I would then add a todo to implement an iterator interface. We can revisit that later.